lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -137,23 +147,20 @@ class TraceElements extends Gatherer {
       backendNodeIds.push(lcpNodeId);
     }
     backendNodeIds.push(...clsNodeIds);
-    // DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds.
-    await driver.sendCommand('DOM.getDocument', {depth: -1, pierce: true});
-    const translatedIds = await driver.sendCommand('DOM.pushNodesByBackendIdsToFrontend',
-      {backendNodeIds: backendNodeIds});
 
     // Mark the elements so we can find them in the page.
     for (let i = 0; i < backendNodeIds.length; i++) {
       const metricName =
         lcpNodeId === backendNodeIds[i] ? 'largest-contentful-paint' : 'cumulative-layout-shift';
-      const nodeDetails = await driver.sendCommand('DOM.describeNode', {nodeId: translatedIds.nodeIds[i]});
-      if (nodeDetails.node.nodeType != 1) {
-        continue;
-      }
-      await driver.sendCommand('DOM.setAttributeValue', {
-        nodeId: translatedIds.nodeIds[i],
-        name: LH_ATTRIBUTE_MARKER,
-        value: metricName,
+      const resolveNodeResponse =
+        await driver.sendCommand('DOM.resolveNode', {backendNodeId: backendNodeIds[i]});
+      const objectId = resolveNodeResponse.object.objectId;
+      await driver.sendCommand('Runtime.callFunctionOn', {
+        objectId,
+        functionDeclaration: `function () {
+          ${setAttributeMarker};
```

Comment:
```suggestion
          ${setAttributeMarker.toString()};
```
 
just to make it explicit

Reply:
gross